Lombiq.NodeJs.Extensions.Samples/package.json

```diff
@@ -1,12 +1,9 @@
 {
   "private": true,
   "devDependencies": {
-    "nodejs-extensions": "link:..\\Lombiq.NodeJs.Extensions",
```

Comment:
@Piedone Removing this as it will be added automatically during the first build.

Reply:
Are these changes due to this? (It's a .patch file but GH doesn't allow it.) Because during build nothing should change in the consumer project that is or might be stored in its repo and is environment specific (these reference my user folder).

[0001-Package-file-change.txt](https://github.com/Lombiq/NodeJs-Extensions/files/8391738/0001-Package-file-change.txt)

Reply:
Yes.

I'm coming to think the installation of NE as an `npm package` will have to be something that consumers will have to do by hand, using an MSBuild target (because we may need to access the `NuGetPackageRoot`). Then, the changes to _package.json_ and `pnpm-lock.yaml` will be self-produced and obviously necessary to commit.

With that we would avoid adjusting those files during build.

Hm, I'm not yet sure how to reference the extension from the NuGet cache in a way that will be user-agnostic 🤔

Reply:
In case of the submodule usage, I can try to use a relative path. For the NuGet package usage, I still have to investigate.

Reply:
We can keep the relative path here. When using the NE NuGet package, though, its location is always kind of environment-specific, as it uses the system's NuGet packages folder. What now?

Reply:
I don't have an existing answer for this. This is one of the fundamental questions why I've opened https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/issues/48 in the first place.

Reply:
I'm thinking that we could copy the NE _package.json_ into a subfolder of the consuming project, and `npm install` it from there - then all of the work will happen inside of that folder. That would make us independent of the NuGet folder structure.

Reply:
I'm OK with that.

Reply:
I've got that working now.

Reply:
Great!

Reply:
The automatic reference now points to `node_modules\.nx`. I've chosen the short `.nx` segment because I've hit a problem when some dependent npm package's installation path hit the 260 character boundary. I now that you can enable support for longer paths on Windows, and I do have that enabled. This part will still fail in that case.

I've added according documentation under the **Troubleshooting** section in the _Readme_.